### PR TITLE
feat(inputs.opcua): add use regular reads workaround

### LINKED ISSUE #11559: Support opcua auto-reconnect and don't use registered reads
One of the gopcua library maintainers pointed out that telegraf needs to handle auto-reconnect and that we should reconsider using registered reads. https://github.com/influxdata/telegraf/issues/9164#issuecomment-1039878729

The related upstream issue is https://github.com/gopcua/opcua/issues/557

### PR DESCRIPTION
**Pull request by R290**
# Required for all PRs

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #11559

Added workaround to use regular reads instead of registered reads.

### CODE REVIEW

File: plugins/inputs/opcua/README.md [hunk 1]
```diff
@@ -152,6 +155,8 @@ This example group configuration has two groups with two nodes each:
   ]
 ```
 
+## Metrics
+
```
Comment (Hipska) [inline plugins/inputs/opcua/README.md:159]:
```suggestion
## Metrics

Metrics are produced according to the defined node ID and group configuration.

```

File: plugins/inputs/opcua/opcua.go [hunk 2]
```diff
@@ -439,7 +454,7 @@ func (o *OpcUA) getData() error {
 	resp, err := o.client.Read(o.req)
 	if err != nil {
 		o.ReadError.Incr(1)
-		return fmt.Errorf("RegisterNodes Read failed: %v", err)
+		return fmt.Errorf("Read failed: %v", err)
```
Comment (Hipska) [inline plugins/inputs/opcua/opcua.go:457]: While refactoring, please also use `%w` for errors:
```suggestion
		return fmt.Errorf("Read failed: %w", err)
```

File: plugins/inputs/opcua/sample.conf [hunk 3]
```diff
@@ -85,3 +85,6 @@
   # [inputs.opcua.workarounds]
     ## Set additional valid status codes, StatusOK (0x0) is always considered valid
     # additional_valid_status_codes = ["0xC0"]
+
+    ## Use regular reads instead of registered reads
+    # use_regular_reads = true
```
Comment (Hipska) [inline plugins/inputs/opcua/sample.conf:90]: The example conf should represent the default values for commented out parameters:
```suggestion
    # use_regular_reads = false

```

**Review by Hipska — CHANGES_REQUESTED** [review]

**Review by Hipska — APPROVED** [review]

**Comment by Hipska** [comment]
Have a look auth the `test-go-linux` check and fix the issues that come from it.
You can also run this on your local machine before pushing, see [Unit Tests](https://github.com/influxdata/telegraf/tree/master/docs/developers#common-development-tasks) section.

**Comment by R290** [comment]
Good point on the `make docs`. I accepted your changes from within GitHub, so I didn't run any checks before committing. For some reason `make check` wants me to change 375 files (including opcua.go) using `make fmt`...

**Comment by powersj** [comment]
@R290 is this one ready to review or was there additional development required? Thanks!

**Comment by R290** [comment]
The attached issue contains of 2 parts, and this pull only resolves the second part:
```telegraf needs to handle auto-reconnect and that we should reconsider using registered reads```
For the first part some issues within the underlying opcua library need to be fixed, but this has been eerily quiet for the past six weeks or so. Feel free to review and merge though, the changes are only a small workaround.

**Review by powersj — APPROVED** [review]
Thanks for putting this up. fwiw I think I held off earlier because this was marked draft.

**Review by MyaLongmire — APPROVED** [review]

**Comment by LarsStegman** [comment]
I know this has already been merged, but it has not yet been released. Might I suggest using `UseUnregisteredReads` as the config name. I think `regular` could be confusing.

**Comment by MyaLongmire** [comment]
@LarsStegman that's a great idea! Are you willing to put up a pr or should i?

**Comment by LarsStegman** [comment]
You can do it if you want 🙂

**Comment by MyaLongmire** [comment]
@LarsStegman pr can be found [here](https://github.com/influxdata/telegraf/pull/11892) :)

**Pull request merged**